internal/controller/pkg/resolver/reconciler.go

```diff
@@ -414,7 +418,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	pkg.SetSource(fmt.Sprintf(format, ref.String(), newVer))
 	if err := r.client.Update(ctx, pkg); err != nil {
 		log.Debug(errUpdateDependency, "error", err)
-		r.record.Event(lock, event.Warning(reasonErrUpdate, err))
+		lock.SetConditions(v1beta1.ResolutionFailed(errors.Wrap(err, errUpdateDependency)))
+		_ = r.client.Status().Update(ctx, lock)
 		return reconcile.Result{}, errors.Wrap(err, errUpdateDependency)
 	}
 
```

Comment:
Shouldn't we set it to success here as well?

```suggestion
    lock.SetConditions(v1beta1.ResolutionSucceeded())
```

Reply:
because we did the resolution even though we couldn't update the package?